nats.go

```diff
@@ -2026,33 +2018,32 @@ func (nc *Conn) Request(subj string, data []byte, timeout time.Duration) (*Msg,
 	}
 
 	// snapshot
-	var doSetup, useOldRequestStyle bool
 	nc.mu.Lock()
-	useOldRequestStyle = nc.Opts.UseOldRequestStyle
-	doSetup = (nc.respMux == nil)
+	useOldRequestStyle := nc.Opts.UseOldRequestStyle
 	nc.mu.Unlock()
 
 	// If user wants the old style.
 	if useOldRequestStyle {
 		return nc.oldRequest(subj, data, timeout)
 	}
 
-	// Make sure scoped subscription is setup at least once on first
-	// call to Request(). Will handle duplicates in createRespMux.
-	if doSetup {
-		if err := nc.createRespMux(); err != nil {
-			return nil, err
-		}
+	// Make sure scoped subscription is setup only once on first call to
+	// Request().
+	var err error
+	nc.respSetup.Do(func() { err = nc.createRespMux() })
```

Comment:
Do with think their is a race using different locks to protect the same structure fields? Should we try a test that spins up some Go routines and then they all try to subscribe at the same time?